tools/nsys_profile.py

```diff
@@ -0,0 +1,116 @@
+"""This script is an e2e tool to start a model run and profile the run.
+
+It parses the analysis produced by onnxruntime/nsys profiling and prints out the result.
+
+Example usage:
+
+    CUDA_VISIBLE_DEVICES="2" python nsys_profile.py --compiler torchscript --model-dir onnx_models/Speech2Text2ForCausalLM --iteration 20
+
+This runs the torchscript model in onnx_models/Speech2Text2ForCausalLM for 20 iterations and profile the run.
+
+The model and test data must be in the same format specified in `examples/bench_model.py`.
+"""
+
+import argparse
+import datetime
+import os
+import subprocess
+
+import profile_analysis
+
+
+def nsys_profile_benchmark(
+    compiler: str, model_dir: str, iteration: int, report_name: str, nsys_path: str
```

Comment:
Was confused about this PYLINT message for a bit, but it seems we should create a `main` function and move everything under `if __name__ == '__main__'` block there.